android/src/main/java/com/thebluealliance/androidclient/interfaces/ModelTable.java

```diff
@@ -1,20 +1,143 @@
 package com.thebluealliance.androidclient.interfaces;
 
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
 import com.thebluealliance.androidclient.models.BasicModel;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * File created by phil on 6/21/14.
  */
-public interface ModelTable<T extends BasicModel> {
+public abstract class ModelTable<T extends BasicModel> {
+
+    private SQLiteDatabase mDb;
+
+    public ModelTable(SQLiteDatabase db){
+        mDb = db;
+    }
+
+    public long add(T in) {
+        mDb.beginTransaction();
+        long ret;
```

Comment:
document what `ret` is, it isn't immediately obvious from reading through the code.